src/tests/integration/test_generic_db_operations.py

```diff
@@ -832,61 +833,286 @@ def test_sql_error_suggests_using_cte(ip_with_dynamic_db, request):
+
+    assert len(first) == 60
+    assert len(second) == 60
+    assert len(third) == 60
+
+
+@pytest.mark.parametrize(
+    "ip_with_dynamic_db",
+    [
+        "ip_with_duckDB_native",
         "ip_with_duckDB",
-        "ip_with_postgreSQL",
     ],
 )
-def test_temp_table(ip_with_dynamic_db, request):
+def test_autocommit_retrieve_existing_resultssets_duckdb_from(
+    ip_with_dynamic_db, request, test_table_name_dict
+):
     ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
-    out = ip_with_dynamic_db.run_cell(
-        """%%sql
-create temp table my_table as select 42;
-select * from my_table;
-"""
-    )
 
-    assert out.error_in_exec is None
-    assert list(out.result) == [(42,)]
+    ip_with_dynamic_db.run_cell("%config SqlMagic.autocommit=True")
+
+    result = ip_with_dynamic_db.run_cell(
+        f'%sql FROM {test_table_name_dict["numbers"]} LIMIT 5'
+    ).result
 
+    another = ip_with_dynamic_db.run_cell(
+        f"%sql FROM {test_table_name_dict['numbers']} LIMIT 5"
+    ).result
 
-def test_results_sets_are_closed(tmp_empty, ip_empty):
-    ip_empty.run_cell("%config SqlMagic.displaylimit = 3")
+    assert len(result) == 5
+    assert len(another) == 5
 
-    ip_empty.run_cell(
-        "%sql sqlite:///test_results_sets_are_closed.db --alias first-conn"
-    )
 
-    ip_empty.run_cell(
-        "%sql sqlite:///test_results_sets_are_closed.db --alias second-conn"
-    )
+CREATE_TABLE = "CREATE TABLE __TABLE_NAME__ (number INT)"
+CREATE_TEMP_TABLE = "CREATE TEMP TABLE __TABLE_NAME__ (number INT)"
+CREATE_TEMPORARY_TABLE = "CREATE TEMPORARY TABLE __TABLE_NAME__ (number INT)"
+CREATE_GLOBAL_TEMPORARY_TABLE = (
+    "CREATE GLOBAL TEMPORARY TABLE __TABLE_NAME__ (number INT)"
+)
 
-    ip_empty.run_cell(
-        """%%sql first-conn
-CREATE TABLE numbers (
-    x INT PRIMARY KEY
-);
 
-INSERT INTO numbers (x)
-VALUES (1), (2), (3), (4), (5), (6), (7), (8), (9), (10);
-"""
+@pytest.mark.parametrize(
+    "ip_with_dynamic_db, create_table_statement",
+    [
+        ("ip_with_postgreSQL", CREATE_TABLE),
+        ("ip_with_postgreSQL", CREATE_TEMP_TABLE),
+        ("ip_with_mySQL", CREATE_TABLE),
+        ("ip_with_mySQL", CREATE_TEMPORARY_TABLE),
+        ("ip_with_mariaDB", CREATE_TABLE),
+        ("ip_with_mariaDB", CREATE_TEMPORARY_TABLE),
+        ("ip_with_SQLite", CREATE_TABLE),
+        ("ip_with_SQLite", CREATE_TEMP_TABLE),
+        ("ip_with_duckDB", CREATE_TABLE),
+        ("ip_with_duckDB", CREATE_TEMP_TABLE),
+        ("ip_with_duckDB_native", CREATE_TABLE),
+        pytest.param(
+            "ip_with_duckDB_native",
+            CREATE_TEMP_TABLE,
+            marks=pytest.mark.xfail(
+                reason="We're executing operations in different cursors"
+            ),
+        ),
+        pytest.param(
+            "ip_with_MSSQL",
+            CREATE_TABLE,
+            marks=pytest.mark.xfail(
+                reason="We need to close all existing result sets for this to work"
```

Comment:
Should we create specific tests for mssql and oracale?

Reply:
we can keep the tests for mssql and oracle here. to fix them I need to make a few changes, I'll work on them in an upcoming PR, for now I wanted to add the tests so we know we have to work on fixing this